File: Pi/pi.py
#!/usr/bin/python

import random
import math

nsteps = 500

nhits = 0
for _ in range(nsteps):
    x = random.uniform(0.0, 1.0)
    y = random.uniform(0.0, 1.0)
    if math.sqrt(x*x+y*y) < 1.0:
        nhits = nhits+1
print("PI = ", 4.0*nhits/nsteps)

File: Pi/pi.kt
// Compiling: kotlinc pi.kt -include-runtime -d pi.jar
// Running:   java -jar pi.jar

fun main(args: Array<String>) {
    val nsteps: Int = 5000

    var nhits: Int = 0
    for(i in 1..nsteps) {
        var x: Double = Math.random()
        var y: Double = Math.random()

        if (x*x+y*y < 1) {
            nhits++
        }
    }

    var pi: Double = 4.0*nhits/nsteps
    println("PI = $pi")
}
